x-pack/plugin/ingest/src/main/java/org/elasticsearch/xpack/ingest/FingerprintProcessor.java

```diff
@@ -0,0 +1,295 @@
+        if (value instanceof ZonedDateTime) {
+            ZonedDateTime zdt = (ZonedDateTime) value;
+            byte[] zoneIdBytes = zdt.getZone().getId().getBytes(StandardCharsets.UTF_8);
+            byte[] zdtBytes = new byte[32 + zoneIdBytes.length];
+            ByteUtils.writeIntLE(zdt.getYear(), zdtBytes, 0);
+            ByteUtils.writeIntLE(zdt.getMonthValue(), zdtBytes, 4);
+            ByteUtils.writeIntLE(zdt.getDayOfMonth(), zdtBytes, 8);
+            ByteUtils.writeIntLE(zdt.getHour(), zdtBytes, 12);
+            ByteUtils.writeIntLE(zdt.getMinute(), zdtBytes, 16);
+            ByteUtils.writeIntLE(zdt.getSecond(), zdtBytes, 20);
+            ByteUtils.writeIntLE(zdt.getNano(), zdtBytes, 24);
+            ByteUtils.writeIntLE(zdt.getOffset().getTotalSeconds(), zdtBytes, 28);
+            System.arraycopy(zoneIdBytes, 0, zdtBytes, 32, zoneIdBytes.length);
+            return zdtBytes;
+        }
+        if (value instanceof Date) {
+            byte[] dateBytes = new byte[8];
+            ByteUtils.writeLongLE(((Date) value).getTime(), dateBytes, 0);
+            return dateBytes;
+        }
+        if (value == null) {
+            return new byte[0];
+        }
+        throw new IllegalArgumentException("cannot convert object of type [" + value.getClass().getName() + "] to bytes");
+    }
+
+    public List<String> getFields() {
+        return fields;
+    }
+
+    public String getTargetField() {
+        return targetField;
+    }
+
+    public ThreadLocal<Hasher> getThreadLocalHasher() {
+        return threadLocalHasher;
+    }
+
+    public byte[] getSalt() {
+        return salt;
+    }
+
+    public boolean isIgnoreMissing() {
+        return ignoreMissing;
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    public static final class Factory implements Processor.Factory {
+
+        public static final String[] SUPPORTED_DIGESTS = { "MD5", "SHA-1", "SHA-256", "SHA-512" };
+
+        static final String DEFAULT_TARGET = "fingerprint";
+        static final String DEFAULT_SALT = "";
+        static final String DEFAULT_METHOD = "SHA-1";
+
+        @Override
+        public FingerprintProcessor create(
+            Map<String, Processor.Factory> registry,
+            String processorTag,
+            String description,
+            Map<String, Object> config
+        ) throws Exception {
+            List<String> fields = ConfigurationUtils.readList(TYPE, processorTag, config, "fields");
+            if (fields.size() < 1) {
+                throw newConfigurationException(TYPE, processorTag, "fields", "must specify at least one field");
+            }
+
+            String targetField = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "target_field", DEFAULT_TARGET);
+            String salt = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "salt", DEFAULT_SALT);
+            byte[] saltBytes = salt.length() > 0 ? toBytes(salt) : new byte[0];
```

Comment:
I would suggest

```suggestion
            byte[] saltBytes = Strings.hasText(salt) ? toBytes(salt) : new byte[0];
```

So we avoid someone typoing `" "` instead of `""` and accidentally salting their food.